extensions/blocks/instagram-gallery/edit.js

```diff
@@ -146,11 +152,27 @@ const InstagramGalleryEdit = props => {
 					label={ __( 'Instagram Gallery', 'jetpack' ) }
 					notices={ noticeUI }
 				>
-					<Button disabled={ isConnecting } isLarge isPrimary onClick={ connectToService }>
+					<Button
+						disabled={ ! isCurrentUserConnected || isConnecting }
+						isLarge
+						isPrimary
+						onClick={ connectToService }
+					>
 						{ isConnecting
 							? __( 'Connecting…', 'jetpack' )
 							: __( 'Connect your Instagram account', 'jetpack' ) }
 					</Button>
+
+					<Notice isDismissible={ false } status="info">
+						{ __(
+							'To connect your Instagram account, you need to link your account to WordPress.com.',
+							'jetpack'
+						) }
+						<br />
+						<Button href={ addQueryArgs( 'admin.php', { page: 'jetpack#/dashboard' } ) } isLink>
```

Comment:
Instead of linking to the dashboard, maybe we could just add the connect button right there, doing something like this?
https://github.com/Automattic/jetpack/blob/a5ec8c3d067113871b59c829a99df492014781ee/_inc/lib/admin-pages/class.jetpack-react-page.php#L246-L248

Reply:
I realize now the "Next steps" section in the PR description is very long. 😝 

tl;dr
I've tried with `build_connect_url` directly: it opens a WPCOM login page, which eventually is supposed to redirect back to the site (redirect URL is the second parameter of the function).
Unfortunately this is not necessarily the case.

<img width="300" alt="Screenshot 2020-04-16 at 18 36 45" src="https://user-images.githubusercontent.com/2070010/79488262-5f4e7c80-8011-11ea-834d-44361d292a0f.png">

There are two "Back" links:
- The secondary, below the form, respects whatever redirect URL I use in `build_connect_url`. 👍 
- The primary one, the big green button, instead always returns to the Calypso plans page. 👎 

In the PR description you can find examples of what I've tried.
I'm not sure if I'm doing something wrong, or if there's something wrong with the login page instead. 🤔

Reply:
hm sorry, I skipped over that part of the description :\

Reply:
> The primary one, the big green button, instead always returns to the Calypso plans page. 👎

That's most likely because of https://github.com/Automattic/wp-calypso/issues/40156

That flow could use some love in general:
https://github.com/Automattic/wp-calypso/issues/24552
https://github.com/Automattic/wp-calypso/issues/8642

Reply:
Ah! Thanks for fetching those out!
I've been juggling a few things and my brain cogs got stuck at some point trying to investigate stuff across one too many repos.

I'll see if there's an obvious quick win in that page.